src/index.js

```diff
@@ -67,7 +67,9 @@ export default function factory({
     if (attachParams) req.params = params;
     let i = 0;
     const len = handlers.length;
-    const loop = async (next) => handlers[i++](req, res, next);
+    const loop = async (next) => Promise.resolve(handlers[i++](req, res, next))
```

Comment:
In this case, if the handler is synchronous (not async) and throws, it will not be able to process the proceeding `.then` / `.catch`

Reply:
Hi, thanks for taking a look and good catch on the error handling discrepancy. I just pushed a change so that errors thrown by sync middleware and async middleware rejecting will both be handled in one spot.